VirtoCommerce.Storefront/Domain/Cart/CartBuilder.cs

```diff
@@ -168,17 +168,26 @@ public virtual Task RemoveItemAsync(string id)
             return Task.FromResult((object)null);
         }
 
-        public virtual Task AddCouponAsync(string couponCode)
+        public virtual Task<bool> AddCouponAsync(string couponCode)
```

Comment:
We don't need return bool value from this method. The code duplication check can be performed on the UI layer or a distinct addition to the cart.coupons collection